### PR TITLE
Add Postgres support to Cook.

### PR DESCRIPTION
**Pull request by scrosby**
## Changes proposed in this PR

- All of the logic to for Cook to communicate to a Postgres database.
- Logic to create a Cook database in Docker.

## Why are we making these changes?
Create a development environment for Cook to talk to Postgres to enable a migration of Cook from Datomic. Does not include any changes to actually use Postgres tables.

### CODE REVIEW

File: scheduler/postgresql/README.txt [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# For cook to function, it must have a postgresql server available. There are two approaches
+# 1. Run postgresql in docker (this script)
```
Comment (dposada) [inline scheduler/postgresql/README.txt:2]: I'm confused about why this is a `.txt` file if it's a script

File: scheduler/postgresql/README.txt [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+# 2. Run a database outside of docker, for, e.g., development.
+
+# For #2, its pretty easy. The main thing you need to do is create the database and a user, set COOK_DB_TEST_PG_DB, PGPASSWORD and COOK_DB_TEST_PG_USER, (or put a :pg-config section in config.edn, for integration tests), and then you can set it up using
```
Comment (dposada) [inline scheduler/postgresql/README.txt:5]:
```suggestion
# For #2, it's pretty easy. The main thing you need to do is create the database and a user, set COOK_DB_TEST_PG_DB, PGPASSWORD and COOK_DB_TEST_PG_USER, (or put a :pg-config section in config.edn, for integration tests), and then you can set it up using
```

File: scheduler/postgresql/sql/init_cook_database.sql [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+-- If you get a crazy error where the two above lines pass, but 'No schema has been selected to
+-- create in' when running the first CREATE TABLE. It can be caused by there being a capital
+-- letter in cook_schema. Schema names are lowercased when created, but case-sensitive when
+-- in the search path.
+
+--- Setup the tables.
+create table resources (
+   resource_name varchar(30) PRIMARY KEY,
+   resource_description text
```
Comment (dposada) [inline scheduler/postgresql/sql/init_cook_database.sql:19]: Should we make this `NOT NULL`?

File: scheduler/postgresql/README.txt (hunk 1 above)
Comment (scrosby) [reply scheduler/postgresql/README.txt:2]: Its not a script. Its a text file with some bash commands available for copy&paste.

Comment (dposada) [reply scheduler/postgresql/README.txt:2]: In that case, thoughts on making this be a Markdown file with properly formatted code blocks?

File: scheduler/postgresql/sql/init_cook_database.sql (hunk 3 above)
Comment (dposada) [reply scheduler/postgresql/sql/init_cook_database.sql:19]: It wasn't clear to me from the file that this was "currently dead" -- should we add a comment in the file explaining what you explained above?

File: scheduler/postgresql/README.txt (hunk 1 above)
Comment (scrosby) [reply scheduler/postgresql/README.txt]: An excellent suggestion. Done.

File: scheduler/postgresql/sql/init_cook_database.sql (hunk 3 above)
Comment (scrosby) [reply scheduler/postgresql/sql/init_cook_database.sql]: Better idea. I'll just strip it from this review.

**Pull request merged**